library/common/engine.cc

```diff
@@ -154,6 +154,19 @@ envoy_status_t Engine::recordGaugeSub(const std::string& elements, uint64_t amou
   return ENVOY_FAILURE;
 }
 
+envoy_status_t Engine::recordHistogramDurationMs(const std::string& elements, uint64_t amount) {
+  if (server_ && client_scope_) {
+    std::string name = Stats::Utility::sanitizeStatsName(elements);
+    server_->dispatcher().post([this, name, amount]() -> void {
+      Stats::Utility::histogramFromElements(*client_scope_, {Stats::DynamicName(name)},
+                                        Histogram::Unit::Milliseconds)
+          .recordValue(amount);
```

Comment:
This is the only method exposed in `histogram.h` and all we really need to assign more values to the collection of data points in the histogram.